cetlvast/suites/unittest/test_variable_length_array_general_allocation.cpp

```diff
@@ -0,0 +1,860 @@
+#endif
+>;
+// clang-format on
+
+TYPED_TEST_SUITE(VLATestsGeneralAllocation, MyTypes, );
+
+// +-------------------------------------------------------------------------------------------------------------------+
+// | TESTS
+// +-------------------------------------------------------------------------------------------------------------------+
+
+TYPED_TEST(VLATestsGeneralAllocation, TestReserve)
+{
+    typename TestFixture::SubjectType subject{TestFixture::make_allocator()};
+
+    ASSERT_EQ(0U, subject.capacity());
+    ASSERT_EQ(0U, subject.size());
+    ASSERT_GE(this->get_expected_max_size(), subject.max_size());
+    subject.reserve(1);
+    ASSERT_LE(1U, subject.capacity());
+    ASSERT_EQ(0U, subject.size());
+    ASSERT_GE(this->get_expected_max_size(), subject.max_size());
+}
+
+// +-------------------------------------------------------------------------------------------------------------------+
+
+TYPED_TEST(VLATestsGeneralAllocation, TestPush)
+{
+    typename TestFixture::SubjectType subject{TestFixture::make_allocator()};
+    ASSERT_EQ(0U, subject.size());
+
+    typename decltype(subject)::value_type x = 0;
+
+    const std::size_t clamped_max = this->clamped_max_size();
+    subject.reserve(clamped_max);
+    for (std::size_t i = 0; i < clamped_max; ++i)
+    {
+        subject.push_back(x);
+
+        ASSERT_EQ(i + 1, subject.size());
+        ASSERT_LE(subject.size(), subject.capacity());
+
+        ASSERT_EQ(x, subject[i]);
+        x = x + 1;
+    }
+}
+
+// +-------------------------------------------------------------------------------------------------------------------+
+
+TYPED_TEST(VLATestsGeneralAllocation, TestPop)
+{
+    typename TestFixture::SubjectType subject{TestFixture::make_allocator()};
+
+    std::size_t clamped_max = this->clamped_max_size(10ul);
+    ASSERT_LE(1U, clamped_max) << "This test requires a max_size of at least 1.";
+    subject.reserve(clamped_max);
+    const std::size_t reserved = subject.capacity();
+    ASSERT_LE(clamped_max, subject.capacity());
+    subject.push_back(1);
+    ASSERT_EQ(1U, subject.size());
+    ASSERT_EQ(1, subject[0]);
+    ASSERT_EQ(1U, subject.size());
+    subject.pop_back();
+    ASSERT_EQ(0U, subject.size());
+    ASSERT_EQ(reserved, subject.capacity());
+}
+
+// +-------------------------------------------------------------------------------------------------------------------+
+
+TYPED_TEST(VLATestsGeneralAllocation, TestShrink)
+{
+    typename TestFixture::SubjectType subject{TestFixture::make_allocator()};
+
+    std::size_t clamped_max = this->clamped_max_size(10ul);
+    ASSERT_LE(1U, clamped_max) << "This test requires a max_size of at least 1.";
+
+    subject.reserve(clamped_max);
+    const auto reserved = subject.capacity();
+    ASSERT_LE(clamped_max, reserved);
+    subject.push_back(1);
+    ASSERT_EQ(1U, subject.size());
+    ASSERT_EQ(1, subject[0]);
+    ASSERT_EQ(1U, subject.size());
```

Comment:
Why repeat the size assertion around the indexing operation? (in previous test too)

Reply:
Probably cruft.